trio/_tools/gen_exports.py

```diff
@@ -104,6 +105,40 @@ def create_passthrough_args(funcdef: ast.FunctionDef | ast.AsyncFunctionDef) ->
     return "({})".format(", ".join(call_args))
 
 
+def run_linters(file: File, source: str) -> str:
+    """Run isort and black on the specified file, returning the new source.
+
+    :raises ValueError: If either failed.
+    """
+    # Black has an undocumented API, but it doesn't easily allow reading configuration from
+    # pyproject.toml, and simultaneously pass in / receive the code as a string.
+    # https://github.com/psf/black/issues/779
```

Comment:
FYI I played around with black's API (just using `format_file_contents` with `Mode` which was just a retyped config file; just line length and preview though) but a dev on black told me to use the CLI interface like you're doing instead :^)